utils/logger/circular.go

```diff
@@ -0,0 +1,51 @@
+// pmm-agent
+// Copyright (C) 2018 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package logger
+
+import (
+	"container/ring"
+	"sync"
+)
+
+type CircularWriter struct {
```

Comment:
Please replace it with `[]string`. Write the next line to index `i`, change index to `(i+1) % len(slice)`
The ring is cool for its extra power, but we don't need it.